### PR TITLE
Cow-ify repository

### PR DESCRIPTION
**Pull request by nlordell**
This PR just "Cow-ifies" the repository. It tries to:
- Replace references to GPv2 into Cow Protocol
- Replace URLs to the new `cow.fi` ones
- Replace issue links to the ones created in this repo.
- Update dependencies to use the github.com/cowprotocol org forks
- Contract vendoring with new repo pacakges (**Note**, the `@cowprotocol/token@1.1.0` package is not deployed on NPM pending https://github.com/cowprotocol/token/pull/1 - feel free to review that PR as well 😄).

### Test Plan

CI - no code or logic changes, just metadata things.

<details><summary>Run the vendoring script to make sure that the contract sources are correct.</summary>

```
% cargo run -p contracts --features bin --bin vendor
[2022-04-07T09:52:59Z INFO  vendor] vendoring contract artifacts to '/Users/nlordell/Developer/cowprotocol/services/crates/contracts/artifacts'
[2022-04-07T09:52:59Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/a3b570a2aa655d4c4941a67e3db6a06fbd72ef09/pkg/deployments/deployed/mainnet/Authorizer.json", query: None, fragment: None })
[2022-04-07T09:52:59Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/a3b570a2aa655d4c4941a67e3db6a06fbd72ef09/pkg/deployments/deployed/mainnet/Vault.json", query: None, fragment: None })
[2022-04-07T09:52:59Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/a3b570a2aa655d4c4941a67e3db6a06fbd72ef09/pkg/deployments/deployed/mainnet/WeightedPoolFactory.json", query: None, fragment: None })
[2022-04-07T09:52:59Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/a3b570a2aa655d4c4941a67e3db6a06fbd72ef09/pkg/deployments/deployed/mainnet/WeightedPool2TokensFactory.json", query: None, fragment: None })
[2022-04-07T09:53:00Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/ad1442113b26ec22081c2047e2ec95355a7f12ba/pkg/deployments/tasks/20210624-stable-pool/abi/StablePoolFactory.json", query: None, fragment: None })
[2022-04-07T09:53:00Z INFO  vendor] retrieving Npm("@openzeppelin/contracts@2.5.0/build/contracts/ERC20Mintable.json")
[2022-04-07T09:53:02Z INFO  vendor] retrieving Npm("@cowprotocol/contracts@1.1.2/deployments/mainnet/GPv2AllowListAuthentication_Implementation.json")
[2022-04-07T09:53:05Z INFO  vendor] retrieving Npm("@cowprotocol/contracts@1.1.2/deployments/mainnet/GPv2Settlement.json")
[2022-04-07T09:53:05Z INFO  vendor] retrieving Npm("@uniswap/v2-core@1.0.1/build/UniswapV2Factory.json")
[2022-04-07T09:53:07Z INFO  vendor] retrieving Npm("@uniswap/v2-periphery@1.1.0-beta.0/build/UniswapV2Router02.json")
[2022-04-07T09:53:08Z INFO  vendor] retrieving Npm("canonical-weth@1.4.0/build/contracts/WETH9.json")
[2022-04-07T09:53:10Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/7a643349a5ef4511234b19a33e3f18d30770cb66/pkg/deployments/tasks/20210721-liquidity-bootstrapping-pool/abi/LiquidityBootstrappingPool.json", query: None, fragment: None })
[2022-04-07T09:53:10Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/7a643349a5ef4511234b19a33e3f18d30770cb66/pkg/deployments/tasks/20210721-liquidity-bootstrapping-pool/abi/LiquidityBootstrappingPoolFactory.json", query: None, fragment: None })
[2022-04-07T09:53:10Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-v2-monorepo/a3b570a2aa655d4c4941a67e3db6a06fbd72ef09/pkg/deployments/extra-abis/WeightedPool.json", query: None, fragment: None })
[2022-04-07T09:53:11Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/balancer-labs/balancer-subgraph-v2/2b97edd5e65aed06718ce64a69111ccdabccf048/abis/StablePool.json", query: None, fragment: None })
[2022-04-07T09:53:11Z INFO  vendor] retrieving Npm("@openzeppelin/contracts@3.3.0/build/contracts/ERC20.json")
[2022-04-07T09:53:15Z INFO  vendor] retrieving Npm("@uniswap/v2-periphery@1.1.0-beta.0/build/IUniswapV2Factory.json")
[2022-04-07T09:53:21Z INFO  vendor] retrieving Npm("@uniswap/v2-periphery@1.1.0-beta.0/build/IUniswapV2Pair.json")
[2022-04-07T09:53:24Z INFO  vendor] retrieving Npm("@uniswap/v2-periphery@1.1.0-beta.0/build/IUniswapV2Router02.json")
[2022-04-07T09:53:25Z INFO  vendor] skipping BalancerV2BasePool: Balancer does not publish ABIs for base contracts
[2022-04-07T09:53:25Z INFO  vendor] skipping BalancerV2BasePoolFactory: Balancer does not publish ABIs for base contracts
[2022-04-07T09:53:25Z INFO  vendor] retrieving Npm("@uniswap/v3-core@1.0.0/artifacts/contracts/interfaces/IUniswapV3Factory.sol/IUniswapV3Factory.json")
[2022-04-07T09:53:26Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/0xProject/protocol/c1177416f50c2465ee030dacc14ff996eebd4e74/packages/contract-artifacts/artifacts/IZeroEx.json", query: None, fragment: None })
[2022-04-07T09:53:26Z INFO  vendor] retrieving Http(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("raw.githubusercontent.com")), port: None, path: "/levelkdev/dxswap-core/3511bab996096f9c9c9bc3af0d94222650fd1e40/build/IDXswapPair.json", query: None, fragment: None })
[2022-04-07T09:53:27Z INFO  vendor] retrieving Npm("@cowprotocol/token@1.1.0/build/artifacts/src/contracts/CowProtocolToken.sol/CowProtocolToken.json")
[2022-04-07T09:53:30Z INFO  vendor] retrieving Npm("@cowprotocol/token@1.1.0/build/artifacts/src/contracts/CowProtocolVirtualToken.sol/CowProtocolVirtualToken.json")
```

</details>

### CODE REVIEW

File: crates/solver/Cargo.toml
Comment (vkgnosis) [inline crates/solver/Cargo.toml:27]:
```suggestion
gas-estimation = { git = "https://github.com/cowprotocol/gas-estimation", tag = "v0.5.0", features = ["web3_"] }
```

**Review by MartinquaXD — APPROVED** [review]
The replacements seem reasonable and the links I tested are working (except the one Valentin already highlighted).
I'm already approving assuming you fix the broken link.

File: crates/solver/Cargo.toml
Comment (nlordell) [reply crates/solver/Cargo.toml:27]: How did this build?

**Review by vkgnosis — APPROVED** [review]

**Comment by nlordell** [comment]
Just holding off merging waiting for cowprotocol/token#1.

**Comment by nlordell** [comment]
Token repo was publised to NPM and I ran the vendor script to make sure it works (see updated test plan).

**Pull request merged**
